magento/magento2
# Issue 30286 - Fix widget layout update removal button to work for multiple layout updates

### Description (*)
When adding a widget in the Magento admin (Content > Widgets), the remove button (trash can icon) on layout updates doesn't work for any layout update but the first. This is because the current code relies on \Magento\Framework\View\Helper\SecureHtmlRenderer::renderEventListenerAsTag to generate the js click listeners, and that uses document.querySelector to establish the element. The element selector passed to querySelector is not unique because it comes from javascript templating, and it always selects the first matching element on the page. Every time the "Add Layout Button" is clicked, another layout row is added, but the click handler is just added to the first row again.

Using a widget block (PHP) class in combination with javascript templating to produce the click listeners doesn't work for this scenario, so this PR updates the remove buttons to use a standard onclick attribute. It also updates the new widget MFTF test to add and remove layout rows and confirm the resulting counts.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30286

### Manual testing scenarios (*)

1. In admin, go to Content > Widgets and click New Widget.
2. Select any widget type and design theme and click Continue.
3. Under Layout Updates, click Add Layout Update multiple times.
4. Click the trash can icon on any of the resulting rows and confirm that the intended row is removed.
5. Give special attention to removing a row other than the first when there are multiples (that was the original failure which prompted this PR).

### Contribution checklist (*)
